sdk/tables/data-tables/src/serialization.ts

```diff
@@ -145,13 +145,51 @@ export function deserialize<T extends object = Record<string, any>>(
       if (`${key}@odata.type` in obj) {
         const type = (obj as any)[`${key}@odata.type`];
         typedValue = getTypedObject(value, type, disableTypeConversion);
+      } else if(["number", "string"].includes(typeof value)) {
+        // The service, doesn't return type metadata for number or strings
+        // if automatic type conversion is disabled we'll infer the EDM object
+        typedValue = inferTypedObject(key, value, disableTypeConversion);
       }
+
       deserialized[transformedKey] = typedValue;
     }
   }
   return deserialized;
 }
 
+function inferTypedObject(propertyName: string, value: number | string, disableTypeConversion: boolean) {
+  // Use value as is when typeConversion is enabled
+  if(!disableTypeConversion) {
+    return value;
+  }
+
+  // We need to skip service metadata fields such as partitionKey and rowKey and use the same value returned by the service
+  if (propertyCaseMap.has(propertyName)) {
+    return value;
+  }
+
+  return typeof value === "string" ? getTypedString(value) : getTypedNumber(value);
```

Comment:
`getTypedString` seems unnecessary

```suggestion
  return typeof value === "string" ? {value, type: "String"} : getTypedNumber(value);
```